src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -355,15 +380,28 @@ export class SavedObjectsRepository {
 
       const method = object.id && overwrite ? 'index' : 'create';
       const requiresNamespacesCheck = object.id && this._registry.isMultiNamespace(object.type);
+      /**
+       * Only when importing an object to a target workspace should we check if the object is workspace-specific.
+       */
+      const requiresWorkspaceCheck = object.id && options.workspaces;
 
       if (object.id == null) object.id = uuid.v1();
```

Comment:
NIT1: Although, undefined == null is true, undefined === null is false. I would suggest:
`
if (!object.id) object.id = uuid.v1();
`
NIT2: Since `requiresNamespacesCheck` and `requiresWorkspaceCheck` are used as boolean variables, I would suggest:
`
const requiresWorkspaceCheck = !!(object.id && options.workspaces);
`

Reply:
NIT2: done